File: 520-detect-capital/detect-capital.py
class Solution(object):
    def detectCapitalUse(self, word):
        """
        :type word: str
        :rtype: bool
        """

        n = len(word)
        countUpper = 0
        countLower = 0

        for idx,char in enumerate(word):
            if char.isupper():
                countUpper += 1
                idx_val = idx

            elif char.islower():
                countLower += 1

        if(countUpper == n or countLower == n):
            return True

        elif (countUpper == 1 and idx_val == 0):
            return True
        
        return False
        

File: 520-detect-capital/detect-capital.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        int n = word.size();
        int countUpper = 0;
        int countLower = 0;
        int idx = -1;

        for(int i = 0; i<n; i++){
            if(isupper(word[i])){
                countUpper++;
                idx = i;
            }
            else if(islower(word[i])){
                countLower++;
            }
        }

        if(countUpper == n || countLower == n){
            return true;
        }
        else if ((countUpper == 1 && idx == 0) || (countUpper == n)){
            return true;
        }
        return false;
    }
};